Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright 2015-2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Apis.Bigquery.v2;
+using Google.Apis.Bigquery.v2.Data;
+using Google.PowerShell.Common;
+using System;
+using System.Management.Automation;
+
+namespace Google.PowerShell.BigQuery
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Creates a new empty dataset in the specified project.
+    /// </para>
+    /// <para type="description">
+    /// Creates a new empty dataset in the specified project.  This command can either take a premade Dataset object, 
+    /// or build one using parameters.  Premade Datasets can be passed via the pipeline or with the -Dataset parameter.
+    /// If both are supplied, the premade dataset object will be used and the builder values will be ignored.  If no 
+    /// Project is specified, the default project will be used.  This cmdlet returns a Dataset object.  
+    /// The builder option will create a new Dataset object and inject all of the supplied values before insertion. 
+    /// Required build parameters are passed directly after the command, whereas optional parameters are passed as flags.
+    /// -- Builder parameters: 
+    /// DatasetId (required) - unique identifier for the dataset.
+    /// Name - descriptive name for the dataset.
+    /// Description - user-friendly description of the dataset.
+    /// Timeout - default duration in ms for tables in the dataset to exist.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> $dataset | New-GcbqDataset -Project $projId </code>
+    ///   <para>This command will take a premade Dataset object from the pipeline and insert it into the Cloud project. 
+    ///   project $projId</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> New-GcbqDataset "test_data_id" -Name "Testdata" -Description "Some interesting data!" -Timeout 86400000</code>
+    ///   <para>This command will build a new dataset with the supplied datasetId, name, description, and a timeout of 1 day.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets)">
+    /// [BigQuery Datasets]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "GcbqDataset")]
+    public class NewGcbqDataset : GcbqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for datasets in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = "premade")]
```

Comment:
DataSet name should be Pascal case. Also, I think you should name it "ByObject" and "ByValues" to be consistent with what we've been using.